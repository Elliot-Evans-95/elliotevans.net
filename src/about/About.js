// @flow

import React, { PureComponent } from 'react';
import { Main } from '../shared/styles/styles';
// import { Helmet, HelmetProvider } from 'react-helmet-async';
import Banner from '../shared/components/Banner';
import AltCard from '../shared/components/AltCard';
import connect from 'react-redux/es/connect/connect';
import { AboutBanner } from './About.style';
import Blobs from '../blobs/Blobs';
import { Meta } from '../meta/meta';

const mapStateToProps = state => ({
  about: state.About
});

class AboutMe extends PureComponent {
  componentDidMount() {
    const title = 'Elliot Evans - About';
    const desc = 'About Page';
    const keywords =
      'Front End Developer, Web Application Developer, Web Developer, Javascript Developer';
    new Meta(title, desc, keywords).setMeta();

    document.getElementById('root').style.backgroundImage = 'url(about.svg)';
  }

  render() {
    return (
      <Main>
        {/*<HelmetProvider>*/}
        {/*<Helmet>*/}
        {/*  <title>Elliot Evans - About</title>*/}
        {/*  <meta name="description" content="About Page" />*/}
        {/*  <meta*/}
        {/*    name="keywords"*/}
        {/*    content="Front End Developer, Web Application Developer, Web Developer, Javascript Developer"*/}
        {/*  />*/}
        {/*</Helmet>*/}
        <Blobs location={this.props.location} />
        <AboutBanner>
          <Banner text={this.props.about.banner} />
        </AboutBanner>
        <AltCard text={this.props.about.card} image={this.props.about.image} />
        {/*</HelmetProvider>*/}
      </Main>
    );
  }
}

const About = connect(mapStateToProps)(AboutMe);

export default About;
