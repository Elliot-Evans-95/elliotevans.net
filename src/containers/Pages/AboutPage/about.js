// @flow

import React from 'react';

import { Helmet } from 'react-helmet';

import { Main } from '../../../assets/styles/styles';
import Card from '../../../components/card';
import Banner from '../../../components/banner';

const About = () => {
  const text = [
    'About me',
    '',
    'Example',
    'There would be a short description from one of my blog posts but instead here is some placeholder text'
  ];

  return (
    <Main>
      <Helmet>
        <title>Elliot Evans - About</title>
        <meta name="description" content="About Page" />
        <meta
          name="keywords"
          content="Software Engineer,Product Manager,Project Manager,Data Scientist,Computer Scientist"
        />
      </Helmet>
      <Banner text={text} />
      <Card text={text} />
    </Main>
  );
};

export default About;
