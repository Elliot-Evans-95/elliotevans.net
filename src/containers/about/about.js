import React from 'react';

import EEBanner from '../../components/banner/ee-banner';
import EECard from '../../components/card/ee-card';
import { Main } from '../../styles';
import PropTypes from 'prop-types';

const About = () => {
  const text = [
    'About me',
    '',
    'Example',
    'There would be a short description from one of my blog posts but instead here is some placeholder text'
  ];

  return (
    <Main>
      <section />
      <EEBanner text={text} />
      <EECard text={text} />
    </Main>
  );
};

About.propTypes = {
  something: PropTypes.object
};

export default About;
