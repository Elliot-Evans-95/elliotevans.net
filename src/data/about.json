{
  "about": "Hello, Elliot here. Thanks for checking out my website. I will hopefully be updating the site bit by bit over ' +\n      'the next few months. In the meantime please checkout some of the blog posts I have made and if you need to' +\n      'contact me then feel free.',\n    'I am currently a web application (fancy way of saying frontend) developer at Codeweavers. I have been at the' +\n      'company for over 2 years crafting great solutions for the motor industry. I have been writing mainly javascript' +\n      'and typescript at the company and using AngularJS and Angular 2+ respectively. Despite this, this website was' +\n      'written in react (I like both frameworks).',\n    'I do fall into the category of trying the new stuff when I can on my personal projects. I also play football and ' +\n      'hockey in my spare time and make music when I feel bored",
  "about2": {
    "1": "",
    "2": "",
    "3": ""
  }
}
