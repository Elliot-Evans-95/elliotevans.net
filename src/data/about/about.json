{
    "pageHeading": "About Me",
    "pageSubHeading": "21yr old Devloper who knows about...",
    "language": [
        {
            "id": "1", 
            "name": "HTML",
            "fullDescription": "lots to talk about here",
            "image": "./html_logo.png",
            "imageDescription": "html logo",
            "experience": "3 years",
            "skills": [
                "HTML5",
                "Semantic",
                "Accessability",
                "Templating"
            ]
        },
        {
            "id": "2", 
            "name": "CSS",
            "fullDescription": "lots to talk about here",
            "image": "./css_logo.png",
            "imageDescription": "css logo",
            "experience": "3 years",
            "skills": [
                "CSS3",
                "CSS Variables",
                "SCSS",
                "PostCSS"
            ]
        },
        {
            "id": "3", 
            "name": "Javascript",
            "fullDescription": "lots to talk about here",
            "image": "./javascript_logo.png",
            "imageDescription": "javascript logo",
            "experience": "3 years",
            "skills": [
                "ES6",
                "Object Oriented programming"
            ]
        },
        {
            "id": "4", 
            "name": "PHP",
            "fullDescription": "lots to talk about here",
            "image": "./php_logo.png",
            "imageDescription": "php logo",
            "experience": "3 years",
            "skills": [
                "RESTful API",
                "MVC",
                "Object Oriented programming"
            ]
        }
    ],
    "framework": [
        {
            "id": "1", 
            "name": "Pug",
            "fullDescription": "lots to talk about here",
            "image": "./pug_logo.png",
            "imageDescription": "pug logo",
            "experience": "2 years",
            "relatedLanguage": [
                "Javascript",
                "html"
            ]
        },
        {
            "id": "2", 
            "name": "Bootstrap",
            "fullDescription": "lots to talk about here",
            "image": "./bootstrap_logo.png",
            "imageDescription": "bootstrap logo",
            "experience": "3 years",
            "relatedLanguage": [
                "CSS",
                "Html",
                "Javascript",
                "Jquery"
            ]
        },
        {
            "id": "3", 
            "name": "AngularJS",
            "fullDescription": "lots to talk about here",
            "image": "./angular_logo.png",
            "imageDescription": "Angular Logo",
            "experience": "1 year",
            "relatedLanguage": [
                "Javascript"
            ]
        },
        {
            "id": "4", 
            "name": "Laravel",
            "fullDescription": "lots to talk about here",
            "image": "./laravel_logo.png",
            "imageDescription": "Larvael Logo",
            "experience": "3 years",
            "relatedLanguage": [
                "PHP"
            ]
        }
    ]
}