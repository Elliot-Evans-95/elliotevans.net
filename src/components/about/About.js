import React, { PureComponent } from 'react';
import { Main } from '../../styles/styles';
import Banner from '../Banner';
import AltCard from '../AltCard';
import { AboutBanner } from './About.style';
import Blobs from '../../blobs/Blobs';
import { Meta } from '../../shared/meta/meta';

const mapStateToProps = state => ({
  about: state.About,
});

class AboutMe extends PureComponent {
  componentDidMount() {
    const title = 'Elliot Evans - About';
    const desc = 'About Page';
    const keywords =
      'Front End Developer, Web Application Developer, Web Developer, Javascript Developer';
    new Meta(title, desc, keywords).setMeta();

    document.getElementById('root').style.backgroundImage = 'url(about.svg)';
  }

  render() {
    return (
      <Main>
        <Blobs location={this.props.location} />
        <AboutBanner>
          <Banner text={this.props.about.banner} />
        </AboutBanner>
        <AltCard text={this.props.about.card} image={this.props.about.image} />
      </Main>
    );
  }
}

export default AboutMe;
