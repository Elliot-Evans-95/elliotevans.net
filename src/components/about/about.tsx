import * as React from 'react';
import { memo } from 'react';
import { Main } from '../../styles/styles';
import { AboutBanner } from './About.style';
import Helmet from 'react-helmet';
import AltCard from '../AltCard';

interface IAbout {
  about: string
}

const About = (about: IAbout) => {
  const title = 'Elliot Evans - About';
  const desc = 'About Page';
  const keywords =
    'Front End Developer, Web Application Developer, Web Developer, Javascript Developer';

    return (
      <Main>
        <Helmet>
          <title>{title}</title>
          <meta name={'description'} content={desc} />
          <meta name={'keywords'} content={keywords} />
        </Helmet>
        <AboutBanner>
          <h1>About</h1>
        </AboutBanner>
        <AltCard text={about}/>
      </Main>
    );
};

export default memo(About);
