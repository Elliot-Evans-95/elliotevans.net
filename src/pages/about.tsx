import * as React from 'react';
import { graphql } from 'gatsby';
import { Main } from '../styles/styles';
import Footer from '../components/footer';

// default style stuff
import '../styles/index.css';

// about stuff
import Navigation from '../components/navigation';
import Blobs from '../blobs/blobs';
import Banner from '../components/banner';
import About from '../components/about/about';

interface IndexPageProps {
  data: {
    allMarkdownRemark: {
      edges: Array<IEdge>;
    };
    allFile: IHeaderQuery;
  };
}

interface IHeaderQuery {
  edges: Array<IHeaderNode>;
}

interface IHeaderNode {
  node: {
    id: string;
    childDataJson: {
      id: string;
      header: IHeader;
      about: string;
    };
  };
}

export interface IEdge {
  node: INode;
}

export interface INode {
  excerpt: string;
  timeToRead: number;
  frontmatter: {
    date: string;
    title: string;
  };
  id: string;
}

export interface IHeader {
  icon: string;
  heading: string;
  subHeading: string;
}

export default class extends React.Component<IndexPageProps, {}> {
  constructor(props: any, context: any) {
    super(props, context);
  }

  // Hook into the edges and create variables for both header and about

  public render() {
    console.log(this.props.data.allFile.edges[0].node.childDataJson);
    return (
      <div className={'appGrid'}>
        <Navigation />
        <React.StrictMode>
          <Main>
            <Blobs props={this.props.location.pathname} />
            <Banner
              header={
                this.props.data.allFile.edges[0].node.childDataJson?.header
              }
            />
            <About
              about={
                this.props.data.allFile.edges[1].node.childDataJson?.about
              }
            />
          </Main>
        </React.StrictMode>
        <Footer />
      </div>
    );
  }
}

export const pageQuery = graphql`
  query {
    allMarkdownRemark {
      totalCount
      edges {
        node {
          id
          frontmatter {
            title
            date(formatString: "DD MMMM, YYYY")
          }
          excerpt
          timeToRead
        }
      }
    }
    allFile(filter: {sourceInstanceName: {eq: "data"}, extension: {eq: "json"}}) {
      edges {
        node {
          id
          childDataJson {
            header {
              icon
              heading
              subHeading
            }
            about
          }
        }
      }
    }
  }
`;
