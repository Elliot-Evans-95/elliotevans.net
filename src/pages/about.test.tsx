import * as React from "react";
import * as renderer from "react-test-renderer";
import { StaticQuery } from "gatsby";
import About from '../components/about/about';
import { IAboutProps} from '../models/about.types';

beforeEach(() => {
  // @ts-ignore
  StaticQuery.mockImplementationOnce(({ render }: IAboutProps) =>
    render({
      site: {
        children: null,
        data: {
          allMarkdownRemark: {
            edges: null
          },
          allFile: null
        },
        location: null,
        pageContext: {
          isCreatedByStatefulCreatePages: null
        },
        path: null,
        pathContext: {
          isCreatedByStatefulCreatePages: null
        },
        uri: null
      },
    })
  )
});

describe("Index", () =>
  it("renders correctly", () => {
    const aboutData = {
      data: {
        "allFile":{
          "edges":[{
            "node":{
              "id":"a2035ede-68a3-5582-9d7e-85ff5b260658",
              "childSiteJson":{
                "id":"64a22821-0894-5392-b2d0-485a03f86a0d",
                "header": "",
                "about":{
                  "paraOne":"Hello, Elliot here. Thanks for checking out my website. I will hopefully be updating the site bit by bit over the year.",
                  "paraTwo":"the next few months. In the meantime please checkout some of the blog posts I have made and if you need to contact me then feel free.",
                  "paraThree":"I am a software developer at bet365 where I have been mainly working of the sports/main site on web and mobile using typescript but I am always trying new languages and frameworks in my spare time.",
                  "paraFour":"Also play hockey every week and make music in between."}
              }
            }
            },
            {
              "node":{
                "id":"2cb785bb-6c88-559b-b4ae-f8279db235ff","childSiteJson":{
                  "id":"e5d6c80b-ea1b-5dca-a9b5-17dc6da60568","header":{
                    "icon":"../assets/images/header-icon.webp",
                    "heading":"Elliot Evans",
                    "subHeading":"Software Developer at Bet365"
                  },
                  "about": ""
                }
              }
            }]
        }
      }
    };

    const tree = renderer.create(<About about={aboutData} />).toJSON();
    expect(tree).toMatchSnapshot()
  }));
