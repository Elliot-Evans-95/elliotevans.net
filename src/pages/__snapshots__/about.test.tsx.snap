// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
<main
  className="css-47ltha-Main e1h04ftx5"
>
  <div
    className="css-19h9wk1-AboutBanner enbx16z0"
  >
    <h1>
      About
    </h1>
  </div>
  <section
    className="css-1ry23gl-AltCardContainer e1h04ftx18"
  >
    <p />
    <p />
    <p />
    <p />
  </section>
</main>
`;
